lib/smack/SmackRep.cpp

```diff
@@ -55,6 +53,26 @@ bool SmackRep::isInt(const llvm::Value* v) {
   return isInt(v->getType());
 }
 
+unsigned SmackRep::getIntSize(const llvm::Value* v) {
+  return getIntSize(v->getType());
+}
+
+unsigned SmackRep::getIntSize(const llvm::Type* t) {
+  const llvm::IntegerType* intType = llvm::cast<llvm::IntegerType>(t);
+  return intType->getBitWidth();
+}
+
+unsigned SmackRep::getElementSize(const llvm::Value* v) {
+  return getElementSize(v->getType()->getPointerElementType());
```

Comment:
The two variants of `getElementSize` are inconsistent, because the first gives the size of the **element type** of the given values type, while the other gives the size of the type itself. Either the selector `getPointerElementType()` should be made inside of `getElementSize(llvm::Type*)`, or `getElementSize(llvm::Type*)` should be renamed to `getSize(llvm::Type*)`.